types/chrome/index.d.ts

```diff
@@ -2,6 +2,9 @@
 /// <reference path="./har-format/index.d.ts" />
 /// <reference path="./chrome-cast/index.d.ts" />
 
+// NoInfer for old TypeScript versions
+type NoInferX<T> = T[][T extends any ? 0 : never];
```

Comment:
IIRC this will actually make the type available globally, no?

Reply:
Hmm.. I think you're correct. That was not my intention.

I guess our options are:

1. Set `"minimumTypeScriptVersion": "5.4"` and use the real `NoInfer`. Maybe this is a tall order.
2. Remove `NoInferX` and its usage, and revert `keys` to be `string | string[]`. The keys won't have any type checking anymore but that was probably the least beneficial thing about this change.

I think 2 might be the right choice for now, and maybe down the line the real `NoInfer` can be used.

Reply:
To be fair, it is a little unfortunate for libs that introduce globals to not be able to cleanly make their own helpers.

Is this type only used within the `chrome.storage` namespace? That _does_ contain explicit exports, so it may be the case that declaring it there would avoid exporting it.

Reply:
I just tested that and it still exposes it but you would have to use `chrome.storage.NoInferX`. So less likely to cause issues or confusion.

I also found that there's an `@internal` JSDoc tag and [`stripInternal`](https://www.typescriptlang.org/tsconfig/#stripInternal) tsconfig setting, but I think it only makes a difference when code is being emitted.

And then I finally found a github comment explaining how to prevent export of all types in `.d.ts` files: https://github.com/microsoft/TypeScript/issues/57764#issuecomment-1995320147

I pushed a new commit that moves `NoInferX`, adds `@internal` for good measure, and then an `export {}` line to stop `NoInferX` from being exported. I had to add `export` to two types as well, because those should be exported.

So I think that should fix this issue. 😄

Reply:
The `export {}` is the key; I would remove `@internal` on the declaration.

Reply:
Ok, done.

Reply:
Just had the idea to add a `@ts-expect-error` test that checks if NoInferX is accidentally exported.